src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -524,3 +533,77 @@ def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_sett
     def do_processes_require_restart(self):
         """Defaulting this for Apt"""
         return False
+
+    def is_reboot_pending(self):
+        """ Checks if there is a pending reboot on the machine. """
+        try:
+            pending_file_exists = os.path.isfile(self.REBOOT_PENDING_FILE_PATH)
+            pending_processes_exists = self.do_processes_require_restart()
+            self.composite_logger.log_debug(" - Reboot required debug flags: " + str(pending_file_exists) + ", " + str(
+                pending_processes_exists) + ".")
+            is_reboot_required = pending_file_exists or pending_processes_exists
+
+            # Use Ubuntu Pro Client api in case of Ubuntu VMs.
+            if not self.__pro_client_prereq_met:
+                self.composite_logger.log_debug("Ubuntu Pro Client pre-requisite not met.")
```

Comment:
[Draft Q] Does this need to be logged here if the logging happens when __pro_client_prereq_met variable is set.